examples/bert/bert_ptq_cpu.json

```diff
@@ -31,18 +31,28 @@
         {
             "name": "glue_mrpc",
             "type": "HuggingfaceContainer",
-            "params_config": {
-                "data_name": "glue",
-                "subset": "mrpc",
-                "split": "validation",
-                "input_cols": [
-                    "sentence1",
-                    "sentence2"
-                ],
-                "label_cols": [
-                    "label"
-                ],
-                "batch_size": 1
+            "load_dataset_config": {
+                "params": {
+                    "data_name": "glue",
+                    "subset": "mrpc",
+                    "split": "validation"
+                }
+            },
+            "pre_process_data_config": {
+                "params": {
```

Comment:
do we really need the "params" key name? it would be better if like
```
            "load_dataset_config": {
                    "data_name": "glue",
                    "subset": "mrpc",
                    "split": "validation"
            },
```

Reply:
the `type` info is needless since the "load_dataset_config" already contains the type info

Reply:
> do we really need the `params` key name? it would be better if like
> 
> ```
>             "load_dataset_config": {
>                     "data_name": "glue",
>                     "subset": "mrpc",
>                     "split": "validation"
>             },
> ```

I agree that params seems redundant. I wasn't sure, what would the config class look like since `type` is required parameter and other are all optional. There is no strict list of what the params can be (think of custom data set/ data container implementation which can take any user arguments).

Reply:
> the `type` info is needless since the `load_dataset_config` already contains the type info

Hmm, I am unsure what you mean config already contains the type info. Please elaborate.

Reply:
I also agree `params` nesting is not needed. But like Hitesh said, `type` is still needed if the user wants to override the default component type.

Reply:
I am open to suggestions on how to remove the `params` node in the hierarchy. Is there a way to define a sub class of `ConfigBase` that can take any key-value pair as argument without the need for a dict member i.e. without being explicitly defined as a member of the class? I guess that would be counter-intuitive to the use of `ConfigBase` itself.

Reply:
Oh yeah, I forgot about the logistics of doing that. 

There is a `ConfigBase` subclass `ConfigWithExtraArgs` https://github.com/microsoft/Olive/blob/3d3277b646c2c5a0fa96bcf629a117aa78423176/olive/common/config_utils.py#L145 where any undefined field gets aggregated into the `extra_args` field. If we use this, then instead of accessing `params` in the backend, we could access `extra_args`.

Reply:
If the changes in this PR look good, I would like to merge this change before making the decorative `params` change. It's modifying too many files, and I don't want to sit on them for too long. Merging will be very hard.

Reply:
sounds good to me! as you said, it's a cosmetic change that only looks different on the user side. In the backend, we need a way to hold the params anyways.

Reply:
Is the `type` required? I don't see it in the example.
what I mean is not change the `DataComponentConfig` and just change the plain json dict to `DataComponentConfig` in validator.

Reply:
example of type override ![image](https://github.com/microsoft/Olive/assets/94929125/51a4b910-bb2d-4d7d-8fe3-2b7fa7049bdd)

Reply:
Even without that params is still required to collect only the parameters that are passed down to the creator function. I noticed that DataSet/DataContainer constructors fail if they are passed unknown arguments (including in kwargs). So, it might not actually be possible to remove the params entirely. If we did, we will still be doing custom include/exclude list to collect only the arguments that are passed down to the constructor (but then what are the other parameters for? discard?).